### DISCUSSION TITLE
A command for graceful kill

### DISCUSSION THREAD

**Discussion by CowBoy4mH3LL** [e-m-b-a/emba#1074] [Ideas]
There may an opportunity to save ongoing state, globals, ENVs, etc. -- then use these for differential restart later.
There may also be an opportunity to release any open resources, descriptors, etc.

PS: I am also seeing some broken terminals (e.g. on VSCODE) when force killing (SIGKILL) -- maybe a VSCODE integrated term thing but a clear of stdout on termination/kill can help, maybe

**Comment by m-1-k-3** [comment]
Yes, we have seen broken terminal also multiple times. Some of the modules are not that nice to the terminal ;) 
Mostly I have seen these issues around the user-mode emulator s115. We need to ensure killing processes all the time in this module. I also think that we have some issues like that in the greppit module s99. If you have an idea on protecting the terminal I am very happy to give it a try.
